src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java

```diff
@@ -0,0 +1,240 @@
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.junit.Test;
+
+public class BigQueryDynamicTableFactoryTest {
+  @Test
+  public void factoryIdentifierTestSuccess() {
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    String factoryIdentfier = bigQueryDynamicTableFactory.factoryIdentifier();
+    String expectedFactoryIdentifier = "bigquery";
+    assertThat(factoryIdentfier).isNotNull();
+    assertThat(factoryIdentfier).isEqualTo(expectedFactoryIdentifier);
+  }
+
+  @Test
+  public void requiredOptionsTestSuccess() {
+
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    List<String> expectedOptions =
+        Arrays.asList(
+            "materializationDataset", "materializationProject", "query", "selectedFields", "table");
+    Set<ConfigOption<?>> configOptionSet = bigQueryDynamicTableFactory.requiredOptions();
+    assertThat(configOptionSet).isNotNull();
+    assertThat(configOptionSet.size()).isEqualTo(5);
+    List<String> options = new ArrayList<String>();
+    Iterator<ConfigOption<?>> configOptionIterator = configOptionSet.iterator();
+    while (configOptionIterator.hasNext()) {
+      ConfigOption configOption = configOptionIterator.next();
+      options.add(configOption.key().toString());
+    }
+    Collections.sort(options);
+    assertTrue(expectedOptions.equals(options));
+  }
+
+  @Test
+  public void optionalOptionsTestSuccess() {
+    List<String> expectedOptions =
+        Arrays.asList(
+            "arrowCompressionCodec",
+            "bqBackgroundThreadsPerStream",
+            "bqEncodedCreateReadSessionRequest",
+            "credentials",
+            "credentialsFile",
+            "defaultParallelism",
+            "filter",
+            "format",
+            "gcpAccessToken",
+            "maxParallelism",
+            "parallelism",
+            "partitionExpirationMs",
+            "partitionField",
+            "partitionRequireFilter",
+            "partitionType",
+            "proxyPassword",
+            "proxyUri",
+            "proxyUsername");
+    List<String> options = new ArrayList<String>();
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    Set<ConfigOption<?>> optionalOptions = bigQueryDynamicTableFactory.optionalOptions();
+    assertThat(optionalOptions).isNotNull();
+    assertThat(optionalOptions.size()).isEqualTo(18);
+    optionalOptions.forEach(
```

Comment:
update the method like `requiredOptionsTestSuccess()`

Reply:
renamed